src/pool/srv_util.c

```diff
@@ -776,8 +790,18 @@ nvme_faulty_reaction(int *tgt_ids, int tgt_cnt)
 		smd_free_pool_info(pool_info);
 	}
 
-	D_DEBUG(DB_MGMT, "Faulty reaction done. tgt_cnt:%d, rc:%d\n",
-		tgt_cnt, rc);
+	/*
+	 * In the case of offline pool recovery, allow successful
+	 * completion of the faulty reaction if at least one online,
+	 * healthy pool is present, and no other errors are present.
+	 */
```

Comment:
Why? That'll cause io error when the offline pool come back online, right? I think we have to make sure all affected pool targets (no matter if it's online or offline) are in DOWN or DOWNOUT before we transit the BIO blobstore state.